### ISSUE TITLE
Error in plotPathway(sce, resultName = "VAM_H_CDF", geneset = hallmark,  :    could not find function "plotPathway"

### ISSUE THREAD

**Issue by rpauly** [compbiomed/singleCellTK#571]
Hi, 
I could not find the  function "plotPathway".
I started with seurat object and convert it into a singlecellexperiment.
```
> sce_rp<-as.SingleCellExperiment(rds_file, assay = NULL)
> sce <- importGeneSetsFromMSigDB(sce_rp, categoryIDs = "H", species = "Homo sapiens")
Wed Mar  9 15:58:14 2022 .. Importing 'H' gene sets (n = 50)
Wed Mar  9 15:58:14 2022 .... Completed 50 gene sets for 1
Wed Mar  9 15:58:14 2022 .. Matching gene sets to 'rownames'
> sce <- runVAM(sce, geneSetCollectionName = "H", useAssay = "logcounts")
gene.weights not specified, defaulting all weights to 1
Computing VAM distances for 50 gene sets, 136048 cells and 2000 genes.
Min set size: 4, median size: 35.5
Computing for gene set 50 of size 38
> hallmark <- "HALLMARK_HYPOXIA"
> plotPathway(sce, resultName = "VAM_H_CDF", geneset = hallmark, groupby = "cluster")
Error in plotPathway(sce, resultName = "VAM_H_CDF", geneset = hallmark,  : 
  could not find function "plotPathway"
```
> packageVersion("singleCellTK")
[1] ‘2.4.0’

```
> sessionInfo()
R version 4.1.0 (2021-05-18)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /usr/local/intel/compilers_and_libraries_2020.2.254/linux/mkl/lib/intel64_lin/libmkl_rt.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] singleCellTK_2.4.0          DelayedArray_0.20.0        
 [3] Matrix_1.3-4                SingleCellExperiment_1.16.0
 [5] SummarizedExperiment_1.24.0 Biobase_2.54.0             
 [7] GenomicRanges_1.46.1        GenomeInfoDb_1.30.0        
 [9] IRanges_2.28.0              S4Vectors_0.32.3           
[11] BiocGenerics_0.40.0         MatrixGenerics_1.6.0       
[13] matrixStats_0.61.0          sceasy_0.0.6               
[15] reticulate_1.24             SeuratObject_4.0.4         
[17] Seurat_4.0.5               

loaded via a namespace (and not attached):
  [1] plyr_1.8.6                igraph_1.2.11            
  [3] lazyeval_0.2.2            GSEABase_1.56.0          
  [5] splines_4.1.0             BiocParallel_1.28.3      
  [7] listenv_0.8.0             scattermore_0.8          
  [9] ggplot2_3.3.5             digest_0.6.29            
 [11] htmltools_0.5.2           fansi_1.0.2              
 [13] memoise_2.0.1             magrittr_2.0.2           
 [15] GSVAdata_1.30.0           tensor_1.5               
 [17] cluster_2.1.2             ROCR_1.0-11              
 [19] limma_3.50.0              Biostrings_2.62.0        
 [21] annotate_1.72.0           globals_0.14.0           
 [23] R.utils_2.11.0            spatstat.sparse_2.1-0    
 [25] colorspace_2.0-2          blob_1.2.2               
 [27] ggrepel_0.9.1             dplyr_1.0.8              
 [29] crayon_1.5.0              RCurl_1.98-1.6           
 [31] jsonlite_1.7.3            graph_1.72.0             
 [33] spatstat.data_2.1-2       survival_3.2-13          
 [35] zoo_1.8-9                 glue_1.6.1               
 [37] polyclip_1.10-0           gtable_0.3.0             
 [39] zlibbioc_1.40.0           XVector_0.34.0           
 [41] leiden_0.3.9              DropletUtils_1.14.2      
 [43] Rhdf5lib_1.16.0           future.apply_1.8.1       
 [45] HDF5Array_1.22.1          msigdbr_7.4.1            
 [47] fishpond_2.0.1            abind_1.4-5              
 [49] scales_1.1.1              DBI_1.1.2                
 [51] edgeR_3.36.0              spatstat.random_2.1-0    
 [53] miniUI_0.1.1.1            Rcpp_1.0.8               
 [55] viridisLite_0.4.0         xtable_1.8-4             
 [57] spatstat.core_2.4-0       dqrng_0.3.0              
 [59] bit_4.0.4                 VAM_1.0.0                
 [61] htmlwidgets_1.5.4         httr_1.4.2               
 [63] RColorBrewer_1.1-2        ellipsis_0.3.2           
 [65] ica_1.0-2                 XML_3.99-0.8             
 [67] pkgconfig_2.0.3           R.methodsS3_1.8.1        
 [69] scuttle_1.4.0             uwot_0.1.11              
 [71] deldir_1.0-6              locfit_1.5-9.4           
 [73] utf8_1.2.2                AnnotationDbi_1.56.2     
 [75] tidyselect_1.1.1          rlang_1.0.1              
 [77] reshape2_1.4.4            later_1.3.0              
 [79] cachem_1.0.6              munsell_0.5.0            
 [81] tools_4.1.0               cli_3.2.0                
 [83] RSQLite_2.2.10            generics_0.1.2           
 [85] ggridges_0.5.3            stringr_1.4.0            
 [87] fastmap_1.1.0             goftest_1.2-3            
 [89] bit64_4.0.5               babelgene_21.4           
 [91] fitdistrplus_1.1-6        purrr_0.3.4              
 [93] RANN_2.6.1                KEGGREST_1.34.0          
 [95] pbapply_1.5-0             future_1.23.0            
 [97] nlme_3.1-153              sparseMatrixStats_1.6.0  
 [99] mime_0.12                 R.oo_1.24.0              
[101] compiler_4.1.0            plotly_4.10.0            
[103] png_0.1-7                 spatstat.utils_2.3-0     
[105] tibble_3.1.6              stringi_1.7.6            
[107] lattice_0.20-45           vctrs_0.3.8              
[109] pillar_1.7.0              lifecycle_1.0.1          
[111] rhdf5filters_1.6.0        BiocManager_1.30.16      
[113] spatstat.geom_2.3-2       lmtest_0.9-39            
[115] RcppAnnoy_0.0.19          data.table_1.14.2        
[117] cowplot_1.1.1             bitops_1.0-7             
[119] irlba_2.3.5               httpuv_1.6.5             
[121] patchwork_1.1.1           R6_2.5.1                 
[123] promises_1.2.0.1          KernSmooth_2.23-20       
[125] gridExtra_2.3             parallelly_1.30.0        
[127] codetools_0.2-18          gtools_3.9.2             
[129] MASS_7.3-54               assertthat_0.2.1         
[131] rhdf5_2.38.0              sctransform_0.3.3        
[133] GenomeInfoDbData_1.2.7    mgcv_1.8-38              
[135] parallel_4.1.0            grid_4.1.0               
[137] rpart_4.1-15              beachmat_2.10.0          
[139] tidyr_1.2.0               DelayedMatrixStats_1.16.0
[141] Rtsne_0.15                shiny_1.7.1
```

**Comment by mvfki** [comment]
Hi, 

Thanks for your feedback. Function `plotPathway` is only available in our latest GitHub release (v2.4.1). You can update your singleCellTK installation in your R console by:

```
install.packages("devtools")
devtools::install_github("compbiomed/singleCellTK")
```

Please feel free to get back to us if it still doesn't work!

Best

**Comment by rpauly** [comment]
Thanks! that worked 👍 
Is there a way to sort by seurat clusters? Also, can I plot the boxplot with no individual dots?
Current command:
plotPathway(sce_rp, resultName = "VAM_H_CDF", geneset = hallmark, groupby = "seurat_clusters", boxplot=TRUE)

Thanks,
Rini

**Comment by mvfki** [comment]
If I understood it correctly. did you mean, for example, when we have numeric cluster labels, by numbers such as 1, 2, 3, ..., you would like to have the violin/boxes in the figure placed in the numeric order? If so, then this is what we will have to improve. @ykoga07 I think this is because we are forcing to run `groupBy <- factor(groupBy, unique(groupBy))` right? Maybe we should add a condition `if (!is.factor(groupBy))` ahead of this. 


For excluding dots in the boxplot, `plotPathway` alone does not support such settings currently, but we've already had the plan to include this feature in the next release. However, I can provide some work-around if you need:
```
plotSCEViolin(inSCE = sce, 
              slotName = "reducedDims", 
              itemName = "VAM_H_CDF", 
              feature = hallmark, 
              groupBy = "seurat_clusters", 
              violin = FALSE, 
              boxplot = TRUE, 
              dots = FALSE)
```
This is the generic function underlying the `plotPathway` function. Notice that the last three arguments are exactly the switches for whether to add the elements.

**Comment by rpauly** [comment]
Thanks..that helps! It would also be really good to have the the numeric cluster labels implemented...t would be so much easier to visualize.

Thanks again,
Rini

**Comment by joshua-d-campbell** [comment]
This has been fixed in version [v2.5.1](https://github.com/compbiomed/singleCellTK/releases/tag/v2.5.1).
